File: tests/complexity/PLDI09_Fig4_2.java
public class PLDI09_Fig4_2 {
    public static void vtrace1(int n, int m, int t, int v1, int v2){}
    public static void main (String[] args) {}
    public static int mainQ(int n, int m){
	assert (n > 0);
	assert (m > 0);
	int v1 = n;
	int v2 = 0;
	int t = 0;
	while(v1 > 0){
	    if (v2 < m) {
		v2++;
		v1--;
	    }else{
		v2=0;
	    }
	    t++;
	}

     
	vtrace1(n, m, t, v1, v2);
	//dig2: l23: -t + v2 <= 0, -m + v2 <= 0, v1 == 0, m*n - m*t + n - v2 == 0, v1 - v2 <= -1
	//Note: cannot find relationship among t and m,n  (annotated n+n/m)
	return 0;
    }

}

File: benchmark/c/complexity/pldi09_fig4_2.c
#include <stdio.h>
#include <stdlib.h>

void vassume(int b){}
void vtrace_post(int n, int m, int tCtr, int x, int y){}

int mainQ(int n, int m){
    vassume(n > 0);
    vassume(m > 0);
    int x = n;
    int y = 0;
    int tCtr = 0;
    while(x > 0){
	if (y < m) {
	    y++;
	    x--;
	}else{
	    y=0;
	}
	tCtr++;
    }

    vtrace_post(n, m, tCtr, x, y);
    //%%%traces: int n, int m, int t, int x, int y
    //dig2: l23: -t + y <= 0, -m + y <= 0, x == 0, m*n - m*t + n - y == 0, x - y <= -1
    //Note: cannot find relationship among t and m,n  (annotated n+n/m)
    return 0;
}


void main(int argc, char **argv){
    mainQ(atoi(argv[1]), atoi(argv[2]));
}
